geo/src/algorithm/simplify.rs

```diff
@@ -87,15 +98,34 @@ where
     if farthest_distance > *epsilon {
         // The farthest index was larger than epsilon, so we will recursively simplify subsegments
         // split by the farthest index.
-        let mut intermediate = compute_rdp(&rdp_indices[..=farthest_index], epsilon);
+        let mut intermediate =
+            compute_rdp::<T, INITIAL_MIN>(&rdp_indices[..=farthest_index], simplified_len, epsilon);
+
         intermediate.pop(); // Don't include the farthest index twice
-        intermediate.extend_from_slice(&compute_rdp(&rdp_indices[farthest_index..], epsilon));
-        intermediate
-    } else {
-        // The farthest index was less than or equal to epsilon, so we will retain only the first
-        // and last indices, resulting in the indices inbetween getting culled.
-        vec![first, last]
+
+        intermediate.extend_from_slice(&compute_rdp::<T, INITIAL_MIN>(
+            &rdp_indices[farthest_index..],
+            simplified_len,
+            epsilon,
+        ));
+        return intermediate;
+    }
+
+    // The farthest index was less than or equal to epsilon, so we will retain only the first
+    // and last indices, resulting in the indices inbetween getting culled.
+
+    // Update `simplified_len` to reflect the new number of indices by subtracting the number
+    // of indices we're culling.
+    *simplified_len -= rdp_indices.len() - 2;
+
+    // If `simplified_len` is now lower than the minimum number of indices needed, then don't
+    // perform the culling and return the original input.
+    if *simplified_len <= INITIAL_MIN {
```

Comment:
Can you imagine a situation where this returns a line much larger than INITIAL_MIN?

I'm not very familiar with this algorithm, but it seems like given a pathological 1000 coord polygon could be simplified in 1 step to 2 points, and then because that's below INITIAL_MIN, we'll instead return the original 1000 coord polygon. But I think I'd expect to get a 4-pt polygon in that case.

Or maybe that's not possible and I am misunderstanding.

Reply:
Agreed we would ideally return a 4 coordinate polygon in that case, but I'm not sure how to modify this algorithm to make that happen, and which is maybe worthy of a separate issue. Unless you know a good solution for this case

Reply:
@michaelkirk How do you feel about opening up a follow-up issue for optimizing in that scenario?

Reply:
I actually don't feel that strongly about it, it's just what occurred to me. I'm happy to see it merged if you think it's an improvement.